### PR TITLE
add rust installation to ci base image

### PR DESCRIPTION
**Pull request by demisx**
# Goal
The goal of this PR is to move rust installation to the CI base image. It also fixes some errors observed after runners have been upgraded to ubuntu `22.04`.

Related #1482

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -306,7 +310,7 @@ jobs:
           echo "runtime_filename_${{matrix.network}}=$release_wasm_filename" >> $GITHUB_OUTPUT
       - name: Install srtool-cli
         run: |
-          cargo install --git https://github.com/chevdor/srtool-cli
+          rustup run stable cargo install --git https://github.com/chevdor/srtool-cli
```
Comment (mattheworris) [inline .github/workflows/release.yml:313]: If the self-hosted runner is ubuntu 22.04 and has glibc ^2.3, then we can use a pre-built binary for srtool-cli/srtool, and not need to install the rust toolchain.

Comment (wilwade) [reply .github/workflows/release.yml:313]: @mattheworris I think this is just a temp step. We are trying to get docker-in-docker working so we can remove the need to have anything special on the self-hosted runner.

Comment (wilwade) [reply .github/workflows/release.yml:302]: That said, yes, it is fine to have both installed on the CI base image and fine to build srtool-cli with stable.

**Review by wilwade — APPROVED** [review]

**Review by mattheworris — APPROVED** [review]
💯

**Pull request merged**